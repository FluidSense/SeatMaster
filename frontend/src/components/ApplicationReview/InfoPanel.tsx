import PanelBase from 'nav-frontend-paneler';
import { Element } from 'nav-frontend-typografi';
import React from 'react';

interface IProps {
  title: string;
  text?: string;
}

const InfoPanel: React.FunctionComponent<IProps> = (props) => {
  const { title, text } = props;
  return (
    <div className={'info-panel'} id={`info-panel-${title.replace(' ', '')}`}>
      <Element>{title}</Element>
      <PanelBase border={true}>{text == null ? '' : text}</PanelBase>
    </div>
  );
};

export default InfoPanel;
