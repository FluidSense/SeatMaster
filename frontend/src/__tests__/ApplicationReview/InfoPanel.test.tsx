import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import InfoPanel from '../../components/ApplicationReview/InfoPanel';

describe('InfoPanel', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <InfoPanel title={'This is a title'} text={'This is a text'} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
