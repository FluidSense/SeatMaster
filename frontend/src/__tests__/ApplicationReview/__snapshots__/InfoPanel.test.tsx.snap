// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPanel renders correctly 1`] = `
<div
  className="info-panel"
  id="info-panel-Thisis a title"
>
  <Element>
    This is a title
  </Element>
  <PanelBase
    border={false}
  >
    This is a text
  </PanelBase>
</div>
`;
